sdv/cag/range.py

```diff
@@ -0,0 +1,365 @@
+from sdv.constraints.utils import (
+    cast_to_datetime64,
+    compute_nans_column,
+    get_datetime_diff,
+    revert_nans_columns,
+)
+from sdv.metadata import Metadata
+
+
+class Range(BasePattern):
+    """Ensure that the `middle_column_name` is between `low_column_name` and `high_column_name`.
+
+    The transformation strategy works the same as the Inequality pattern but with two
+    columns instead of one. We compute the difference between the `middle_column_name`
+    and the `low_column_name` column and then apply a logarithm to the difference + 1 to ensure
+    that the value stays positive when reverted afterwards using an exponential.
+    We do the same for the difference between the `high_column_name` and `middle_column_name`.
+
+    Args:
+        low_column_name (str):
+            Name of the column which will be the lower bound.
+        middle_column_name (str):
+            Name of the column that has to be between the lower bound and upper bound.
+        high_column_name (str):
+            Name of the column which will be the higher bound.
+        strict_boundaries (bool, optional):
+            Whether the comparison of the values should be strict `>=` or
+            not `>` when comparing them.
+            Defaults to True.
+        table_name (str, optional):
+            The name of the table that contains the columns. Optional if the
+            data is only a single table. Defaults to None.
+    """
+
+    @staticmethod
+    def _validate_init_inputs(
+        low_column_name,
+        middle_column_name,
+        high_column_name,
+        strict_boundaries,
+        table_name,
+    ):
+        if not (
+            isinstance(low_column_name, str)
+            and isinstance(middle_column_name, str)
+            and isinstance(high_column_name, str)
+        ):
+            raise ValueError(
+                '`low_column_name`, `middle_column_name` and `high_column_name` must be strings.'
+            )
+
+        if not isinstance(strict_boundaries, bool):
+            raise ValueError('`strict_boundaries` must be a boolean.')
+
+        if table_name and not isinstance(table_name, str):
+            raise ValueError('`table_name` must be a string or None.')
+
+    def __init__(
+        self,
+        low_column_name,
+        middle_column_name,
+        high_column_name,
+        strict_boundaries=True,
+        table_name=None,
+    ):
+        super().__init__()
+        self._validate_init_inputs(
+            low_column_name,
+            middle_column_name,
+            high_column_name,
+            strict_boundaries,
+            table_name,
+        )
+        self._low_column_name = low_column_name
+        self._middle_column_name = middle_column_name
+        self._high_column_name = high_column_name
+        self._low_diff_column_name = f'{self._low_column_name}#{self._middle_column_name}'
+        self._high_diff_column_name = f'{self._middle_column_name}#{self._high_column_name}'
```

Comment:
do we need to check these names for collisions? If we don't do that for the constraint, we don't need to here either

Reply:
These variables are used with the `_create_unique_name` method, like [here](https://github.com/sdv-dev/SDV/pull/2407/files#diff-03e714e20c666f9d2f2ffd5a5ba128b443fe144e1037ccb0aaf2d492925c6f34R183), so there are no collisions.